packages/workers-shared/asset-worker/src/index.ts

```diff
@@ -1,56 +1,94 @@
 import { WorkerEntrypoint } from "cloudflare:workers";
 import { AssetsManifest } from "./assets-manifest";
+import { applyConfigurationDefaults } from "./configuration";
+import { getIntent, handleRequest } from "./handler";
 import {
 	InternalServerErrorResponse,
 	MethodNotAllowedResponse,
-	NotFoundResponse,
-	OkResponse,
 } from "./responses";
-import { getAdditionalHeaders, getMergedHeaders } from "./utils/headers";
 import { getAssetWithMetadataFromKV } from "./utils/kv";
+import type { AssetConfig } from "../../utils/types";
+
+type Env = {
+	// ASSETS_MANIFEST is a pipeline binding to an ArrayBuffer containing the
+	// binary-encoded site manifest
+	ASSETS_MANIFEST: ArrayBuffer;
+
+	// ASSETS_KV_NAMESPACE is a pipeline binding to the KV namespace that the
+	// assets are in.
+	ASSETS_KV_NAMESPACE: KVNamespace;
+
+	CONFIG: AssetConfig;
+};
 
 export default class extends WorkerEntrypoint<Env> {
-	async fetch(request: Request) {
+	async fetch(request: Request): Promise<Response> {
 		try {
-			return this.handleRequest(request);
+			return handleRequest(
+				request,
+				applyConfigurationDefaults(this.env.CONFIG),
+				this.exists.bind(this),
+				this.getByETag.bind(this)
```

Comment:
With RPC, it's now quite easy to accidentally expose a method now that you otherwise didn't intend to. I was trying to slim down `index.ts` to exactly the public contract so that it was clear and we didn't lose sight of that in amongst all the logic of `handleRequest`, so yeah, moved that out into a separate place.